### ISSUE TITLE
Premature end of JPEG file

### ISSUE THREAD

**Issue by easy-and-simple** [tensorflow/tensorflow#41208]
Premature end of JPEG file
this is brain bug of tf team that can't imagine that if there is issue with image it should not be processed and should be skipped because after this error optimisation become some mess.

I get this error frequently then I try to train on some basic dataset from images downloaded from internet
Images are processed ad handled by OS and image software without problems and errors
I am using TF2.1 on conda dor windows but my problem is not that it get errors from time to time, but that TF team are not smart enough to:
1. print the path of file giving this error
2. check file health before to use it for training and if it is bad to skip it!!! or better optionally move it to some debug folder from where people could get it and inspect and analyze it, then fix it and return it back
As you understand for this issue, it is not important reason, but is important issue handling that is completely missing
error message Premature end of JPEG file is not handling of anything especially because in this case for normal people it is clear what must be done on this error

**Comment by amahendrakar** [comment]
@easy-and-simple,
Could you please provide a sample code to reproduce the issue reported here, so that we can look into it. Thanks!

**Comment by mihaimaruseac** [comment]
Hi. At the time the error is encountered, we only have a binary representation of the file contents. PRs to change the architecture/pass in more information are welcome. Or RFCs proposing a better design.

Please be respectful, consider the [code of conduct](https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md). We are all human and make design mistakes

**Comment by easy-and-simple** [comment]
STRDYSs are realy JFGETRD
Because WTDPPR right?!

**Comment by easy-and-simple** [comment]
> @easy-and-simple,
> Could you please provide a sample code to reproduce the issue reported here, so that we can look into it. Thanks!

I started with ML and AI about 2 months ago, I am new to all this, and I can't write complicated code, so I am using code that is from github project implementing yolo2 on tensorflow.
This is more detailed error message:
Error 1:
Model.fit_generator (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
Please use Model.fit, which supports generators.

Error 2:
W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Cancelled: Operation was cancelled
Premature end of JPEG file


Code implementing batch generator as whole project is for tensorflow v1, but is updated to tensorflow v2 using tensorflow v2 converter script and some manual changes. In general it is working but this JPEG issue is making problems because it obviously impact training algorithm by making big errors when problematic image is processed. I really doubt that problem is in image because windows process all images without errors and I make thumbnails for them

`class BatchGenerator(Sequence):
    def __init__(self,
                 netin_gen,
                 netout_gen,
                 yolo_box,
                 img_aug,
                 annotations,
                 batch_size,
                 repeat_times):
        """
        # Args
            annotations : Annotations instance
        
        """
        self._netin_gen = netin_gen
        self._netout_gen = netout_gen
        self._img_aug = img_aug
        self._yolo_box = yolo_box

        self._batch_size = min(batch_size, len(annotations)*repeat_times)
        self._repeat_times = repeat_times
        self.annotations = annotations
        self.counter = 0

    def __len__(self):
        return int(len(self.annotations) * self._repeat_times /self._batch_size)

    def __getitem__(self, idx):
        """
        # Args
            idx : batch index
        """
        x_batch = []
        y_batch= []
        for i in range(self._batch_size):
            # 1. get input file & its annotation
            fname = self.annotations.fname(self._batch_size*idx + i)
            boxes = self.annotations.boxes(self._batch_size*idx + i)
            labels = self.annotations.code_labels(self._batch_size*idx + i)
            
            # 2. read image in fixed size
            img, boxes = self._img_aug.imread(fname, boxes)

            # 3. grid scaling centroid boxes
            norm_boxes = self._yolo_box.trans(boxes)
            
            # 4. generate x_batch
            x_batch.append(self._netin_gen.run(img))
            y_batch.append(self._netout_gen.run(norm_boxes, labels))

        x_batch = np.array(x_batch)
        y_batch = np.array(y_batch)
        self.counter += 1
        return x_batch, y_batch

    def on_epoch_end(self):
        self.annotations.shuffle()
        self.counter = 0
`

**Comment by amahendrakar** [comment]
@easy-and-simple, 
> Error 1:
> Model.fit_generator (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
> Instructions for updating:
> Please use Model.fit, which supports generators.

This issue has already been addressed here [#38938](https://github.com/tensorflow/tensorflow/issues/38938) and has been fixed in TF v2.3.0rc1.


> 
> Error 2:
> W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Cancelled: Operation was cancelled
> Premature end of JPEG file
> 
Similarly, this _warning_ too has been addressed in multiple issues previously. Please take a look at these duplicates [#35100](https://github.com/tensorflow/tensorflow/issues/35100#issuecomment-603922186), [#37515](https://github.com/tensorflow/tensorflow/issues/37515#issuecomment-602901519), [#34156](https://github.com/tensorflow/tensorflow/issues/34156#issuecomment-638683792), [#37324](https://github.com/tensorflow/tensorflow/issues/37324#issuecomment-595869309).

Hope this helps. Please feel free to close the issue if resolved. Thanks!

**Comment by easy-and-simple** [comment]
Are you in mind?!

You losers dont teach me and fix this bug!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Issue is with Premature end of JPEG file you dumb morons!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Obviously for google work only morons!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
I hope this farm for real morons google will bankrupt soon

**Comment by lvenugopalan** [comment]
@easy-and-simple Please read [code of conduct](https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md) before posting any queries and be respectful as advised in earlier post. The issue tracker is for Tensorflow feature requests or bug reports. Please refrain from posting inappropriate comments. If any functionality or feature does not work to your satisfaction, please let us know, as each feedback is valuable to us. Tensorflow team will be happy to address them. Were you able to go over the information provided by @amahendrakar. Again, we can only help if you restrict your posts to product issues and not personal rants.

**Comment by shehroz010** [comment]
Is this issue is fixed or not? If someone is asking a question you should provide the exact resolution not some reference link to other issue on which this issues is subpart of. M using TF-GPU 2.3.1 as object detection and I am having the same issue while training the model. So, is there exist any resolution of this issue or not ? If exist what is the changes i need to make in the TF package because i just created record file and fed it to the model and it pops me this error.

**Comment by mihaimaruseac** [comment]
Please open new issue and fill in issue template, provide a minimal reproducer.

**Comment by mnbrshd** [comment]
This issue arises if your JPEG file isn't written or saved properly e.g. if your script is running and you interrupt it while cv2.imwrite or anything similar is being run at the moment and the image gets saved right before the script stops, you'll get this error if you use that image.
